### DISCUSSION TITLE
Support for bytes like object in File class

### DISCUSSION THREAD

**Discussion by infohash** [litestar-org/litestar#1868] [Ideas]
File class takes path of the file on the disk. This assumes that the file to be returned in the [File Response](https://docs.litestar.dev/2/usage/responses.html#file-responses) already exists. However, if you are exporting some data from the database in the form of csv, a bytes like object of the dataframe is enough to send it as a file. While, it can be done with using generic `Response` class, its OpenAPI schema doesn't show that it returns a file object.

## Improvement

Just like `pandas.Dataframe.to_csv` `path_or_buf` argument also accepts bytes buffer as the parameter, we can have the same functionality here in `File`. What do you think?

**Comment by Goldziher** [comment]
i dont see the use case tbh. are there any other interested parties?

**Reply by mike0sv** [reply]
yes. any use-case when you want to generate file dynamically in-memory

**Reply by provinzkraut** [reply]
But why would you want to use the file response then? It's specifically designed to serve files from a file system. If you already have the bytes at hand, why not just send those directly?

**Reply by mike0sv** [reply]
I want response to look like file. For now I am using 
```
def ..... -> Reponse:
  ...
  return Response(
        buf.getvalue(),
        headers={"content-disposition": f"attachment;filename=..."},
    )
```
    
Is there a better way?

**Reply by provinzkraut** [reply]
> Is there a better way?

Is there something wrong with your way?

You could even do something like this if you want to reuse it:

```python
class InMemoryFileResponse(Response[bytes]):
    def __init__(self, buf: IO, name: str) -> None:
        super().__init__(
            buf.getvalue(), 
            headers={"content-disposition": f"attachment;filename={name}"
        )


@get("/")
async def handler() -> InMemoryFileResponse:
  buf = get_buffer()
  return InMemoryFileResponse(buf, name="something")
```
